### PR TITLE
Delete Environment reflection hack

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by stephentoub** [comment]
Thanks!  LGTM, but @ericstj/@weshaggard should verify the packaging/project changes.

**Comment by jkotas** [comment]
@dotnet-bot test Innerloop CentOS7.1 Debug Build and Test  please

File: src/System.Runtime.Extensions/src/System.Runtime.Extensions.builds [hunk 1]
```diff
@@ -9,14 +9,6 @@
...
     <Project Include="System.Runtime.Extensions.csproj">
-      <TargetGroup>netstandard1.5</TargetGroup>
-      <OSGroup>Unix</OSGroup>
-    </Project>
-    <Project Include="System.Runtime.Extensions.csproj">
-      <TargetGroup>netstandard1.5</TargetGroup>
-      <OSGroup>Windows_NT</OSGroup>
-    </Project>
-    <Project Include="System.Runtime.Extensions.csproj">
       <TargetGroup>net462</TargetGroup>
```
Comment (weshaggard) [inline src/System.Runtime.Extensions/src/System.Runtime.Extensions.builds]: I assume your expectation is that the netstandard1.5 implementation will come from the package harvesting? If so I suspect you can also clean out the net462 build configuration similarly.

Comment (jkotas) [inline src/System.Runtime.Extensions/src/System.Runtime.Extensions.builds]: Yes ... fixed.

Comment (jkotas) [inline src/System.Runtime.Extensions/src/System.Runtime.Extensions.builds]: When I remove net462 config, I am running into packaging errors that I do not know how to fix:

```
packageIndex.json is has an assembly version entry(s) for 4.1.1.0 which are no longer in package System.Runtime.Extensions.  Please run /t:UpdatePackageIndex on this project to commit an update.
```

Running `/t:UpdatePackageIndex` does nothing to the `packageIndex.json`.

**Comment by jkotas** [comment]
@dotnet-bot test Innerloop Windows_NT Release Build and Test please

**Comment by jkotas** [comment]
@dotnet-bot test Innerloop CentOS7.1 Debug Build and Test please

**Comment by jkotas** [comment]
@dotnet-bot test Innerloop Ubuntu14.04 Debug Build and Test please

File: src/System.Runtime.Extensions/src/System.Runtime.Extensions.builds (hunk 1 above)
Comment (weshaggard) [inline src/System.Runtime.Extensions/src/System.Runtime.Extensions.builds]: Did you also remove the entry in the pkgproj?

Comment (jkotas) [inline src/System.Runtime.Extensions/src/System.Runtime.Extensions.builds]: Yes. Here is the change https://github.com/jkotas/corefx/commit/891255bb1ad193bac319474ce4d2a864f9c39209 that gives me the error.

Comment (weshaggard) [inline src/System.Runtime.Extensions/src/System.Runtime.Extensions.builds]: @ericstj are there cases were we need to manually remove entries from the packageIndex.json file?

Comment (ericstj) [inline src/System.Runtime.Extensions/src/System.Runtime.Extensions.builds]: Sometimes with split packages.  Try first running it on the win/unix projects  (after building them) then on the root project.

Comment (jkotas) [inline src/System.Runtime.Extensions/src/System.Runtime.Extensions.builds]: Running it on win/unix projects first did update the packageIndex.json (https://github.com/jkotas/corefx/commit/07f7023394bf214120a84a3a1bcedda4a9b6d1f7), but the above error is still there when building the root project

**Pull request merged**